lib/msf/util/exe.rb

```diff
@@ -541,18 +541,18 @@ def self.exe_sub_method(code,opts ={})
       end
       pe[136, 4] = [rand(0x100000000)].pack('V') unless opts[:sub_method]
     when :dll
-      max_length = 4096
+      opts[:exe_max_sub_length] ||= 4096
     when :exe_sub
-      max_length = 4096
+      opts[:exe_max_sub_length] ||= 4096
     end
 
     bo = self.find_payload_tag(pe, "Invalid PE EXE subst template: missing \"PAYLOAD:\" tag")
 
-    if code.length <= max_length
+    if code.length <= opts.fetch(:exe_max_sub_length)
```

Comment:
Should this not have a default value to fall back onto in case none of the case options are hit?

Reply:
I don't think so. I'm not sure what default would make sense other than maybe 4096 I guess. The value will be defined if `opts[:exe_type]` is one of `:service_exe`, `:dll`, `:exe_sub`. Prior to these changes, `max_length` would have been nil  if `opts[:exe_type]` was not one of those values and this would have crashed with `ArgumentError: comparison of Integer with nil failed`. That's a lot less clear to trouble shoot than this which will raise `KeyError: key not found: :exe_max_sub_length`.